lib/fs.js

```diff
@@ -196,13 +196,76 @@ function Stats(
   this.ino = ino;
   this.size = size;
   this.blocks = blocks;
-  this.atime = new Date(atim_msec);
-  this.mtime = new Date(mtim_msec);
-  this.ctime = new Date(ctim_msec);
-  this.birthtime = new Date(birthtim_msec);
+  this._atim_msec = atim_msec;
+  this._mtim_msec = mtim_msec;
+  this._ctim_msec = ctim_msec;
+  this._birthtim_msec = birthtim_msec;
 }
 fs.Stats = Stats;
 
+Object.defineProperties(Stats.prototype, {
+  atime: {
+    configurable: true,
+    enumerable: true,
+    get() {
+      return this._atime !== undefined ?
+          this._atime :
+          (this._atime = new Date(this._atim_msec + 0.5));
+    },
```

Comment:
Hey, yeah I have tried these variations.

Unfortunately a single `defineProperty` is more expensive than 4 date creations, so you'd lose the benefits of the laziness if you need to access even a single date.

`this[somevariable]` is slow enough compared to `this.mtime` that the difference becomes pretty significant in a function like stat, that isn't very heavy to begin with. That drops accessing one time-field twice (my test is `if (stat.ctime.getTime() !== stat.ctime.getTime()) throw err;`) to a 65% improvement over the non-lazy code, down from a 105% improvement if done the ugly way. But I can definitely change to that if it's preferable.. I didn't write it this way because I think it's beautiful. :)